Include/internal/pycore_moduleobject.h

```diff
@@ -22,6 +22,9 @@ typedef struct {
     PyObject *md_weaklist;
     // for logging purposes after md_dict is cleared
     PyObject *md_name;
+#ifdef Py_GIL_DISABLED
+    void *md_gil;
+#endif
```

Comment:
There are alternatives to adding this to `PyModuleObject`, which would be better long-term.  However, that doesn't need to be addressed in this PR.